### PR TITLE
Add setting to hide topics from video feeds

### LINKED ISSUE #271: Feature request: Hide certain topics from video feed
There are certain video types I'm rarely interested in (Apex or ARK, for example), but they're popular enough that my favorites feed is usually full of them. It would be nice if I could filter those out of my feed so I could see the stuff I'm interested in more easily.

The "topics" feature works great, so it would be perfect if we could use that in the other direction to hide videos that match a topic. Maybe it could be controlled with a new field in the "Video Feed Settings" section: select a list of topics, then exclude those from the favorites feed.

### PR DESCRIPTION
**Pull request by ajhall**
Add a setting to hide topics from video feeds. Videos with matching topics are hidden from the Live and Archive views of the Home and Favorites pages.

Move the "hide collabs" filter into the VideoCardList component, which makes it work on the Archive view of the Favorites page.

This is my first time working with Vue and my first time in a while doing any web frontend work, so please let me know if you'd like me to change how I implemented anything.

Closes #271.

### CODE REVIEW

**Comment by RiceCakess** [comment]
Hey thanks for the PR! I think instead of adding the filter to every single api request location. I suggest looking at how we are doing blockedChannels and filtering at the VideoList component level 

https://github.com/HolodexNet/Holodex/blob/e3d6f443046cc98ac7c2248d796d33b271d681a9/src/components/video/VideoCardList.vue#L147

Should simplify the code/change set a bit, let met know if you have any questions. (Maybe also consider joining discord!)

**Comment by ajhall** [comment]
Ahh perfect, that's much simpler. I was basing my changes on the "hide collabs" feature, so I didn't think to use the video list component.

Speaking of that, should the collab filter be moved into the video list as well? Right now it's in favorites.module.js and it only works on the Live tab, not the Archive tab. If it doesn't make sense in the video list (maybe since it's only relevant to favorites) I can just make a separate issue for it.

I don't hang out in Discord much in general, but if I find some time to do any future work on Holodex I'll jump in there!

**Comment by RiceCakess** [comment]
Yea sure go for it! Just need to be more careful, maybe have prop hideCollabs to enable/disable that filter and then enable it where needed.

**Comment by ajhall** [comment]
Cool, sounds good! I made that change as well as another adjustment so ignored topics will only get filtered out on the Home and Favorites pages instead of everywhere (channel pages, etc.). This is ready for another look.

File: src/components/video/VideoCardList.vue [hunk 1]
```diff
@@ -146,17 +154,35 @@ export default {
...
             const blockedChannels = this.$store.getters["settings/blockedChannelIDs"];
+            const ignoredTopics = this.$store.getters["settings/ignoredTopics"];
+            const favoriteChannels = this.$store.getters["favorites/favoriteChannelIDs"];
+
+            const filterVideos = (v) => {
+                let keep = true;
+                const channelId = v.channel_id || v.channel.id;
+
+                if (!this.ignoreBlock) {
+                    keep &&= !blockedChannels.has(channelId);
```
Comment (RiceCakess) [inline src/components/video/VideoCardList.vue:165]: Neat!

**Review by RiceCakess — APPROVED** [review]

**Comment by RiceCakess** [comment]
Thanks for the contribution!

**Pull request merged**